pallets/gear/src/manager.rs

```diff
@@ -325,6 +338,7 @@ where
                 .unwrap_or_else(|e| unreachable!("Message queue corrupted! {:?}", e));
         }
 
+        let _ = common::waiting_init_take_messages(id_exited);
```

Comment:
That won't work.
`waiting_init_take_messages` goes directly to sp_io to already removed storage. Messages from waitlist removes two lines above, but there is still storage for ProgramId and Vec<MessageId> mapping "waiting_init" directly in sp_io.
we need just to clear that prefix - provide such method in common
like that

`# gear-common/src/lib.rs`
```rust
pub fn clear_waiting_init_ids(prog_id: ProgramId) {
    let key = waiting_init_prefix(prog_id);
    sp_io::storage::clear(&key);
}
```